colossalai/fx/codegen/activation_checkpoint_codegen.py

```diff
@@ -307,8 +342,27 @@ def emit_code_with_nested_activation_checkpoint(body, ckpt_func, nodes, emit_nod
         # process node in forward function
         else:
             node = node_list[node_idx]
-            emit_node_func(node, body)
-            delete_unused_value_func(node, body)
+
+            # if a node is outside of checkpoint region and want to offload
+            # it's input activation, we will use torch.saved_tensors_hooks
+            # to complete the offload process.
+            if getattr(node, "activation_offload", False):
+                if not is_hook_inserted:
+                    pack_hook, unpack_hook = _gen_saved_tensors_hooks()
+                    ckpt_func.insert(0, "\n".join([pack_hook, unpack_hook]) + "\n")
+
+                for par in node._input_nodes:
```

Comment:
I think `node._input_nodes` should be `node.all_input_nodes`

Reply:
Sure it will be better~

Reply:
fixed